libs/common/src/schemas/portal-amendment.ts

```diff
@@ -0,0 +1,28 @@
+import z from 'zod';
+import { CURRENCY } from '../constants';
+import { Currency } from '../types';
+import { UNIX_TIMESTAMP_SECONDS_SCHEMA } from './unix-timestamp.schema';
+
+/**
+ * Portal Amendment schema to validate an object contains only user provided values on `PortalFacilityAmendment`
+ *
+ * If this schema is changed the Open Api docs should be updated too
+ */
+export const PORTAL_FACILITY_AMENDMENT = z
+  .object({
+    changeCoverEndDate: z.boolean().optional(),
+    coverEndDate: UNIX_TIMESTAMP_SECONDS_SCHEMA.optional(),
+    currentCoverEndDate: UNIX_TIMESTAMP_SECONDS_SCHEMA.optional(),
+    isUsingFacilityEndDate: z.boolean().optional(),
+    facilityEndDate: UNIX_TIMESTAMP_SECONDS_SCHEMA.optional(),
+    bankReviewDate: UNIX_TIMESTAMP_SECONDS_SCHEMA.optional(),
+    changeFacilityValue: z.boolean().optional(),
+    value: z.number().optional(),
+    currentValue: z.number().optional(),
+    currency: z.enum(Object.values(CURRENCY) as [Currency] & Currency[]).optional(),
+    ukefExposure: z.number().optional(),
+    coveredPercentage: z.number().optional(),
+  })
+  .strict();
+
+Object.values(CURRENCY);
```

Comment:
What is this for, doesn't seem to be doing anything?

Reply:
This throws an error if there are extra properties on the object - by default zod will just strip these out & not throw an error.

I think I prefer returning a 400 than ignoring them - e.g. if the user sends the following body it would be surprising for those values not to be included if a 200 is sent

```
{
 type: 'TFM',
 updatedAt: 1700940940,
 value: 180000
}
```

Reply:
I've just realised I deleted the typo `Object.values(CURRENCY);` you commented on & thought this was in reference to `.strict()`. My reply makes absolutely no sense 🤦